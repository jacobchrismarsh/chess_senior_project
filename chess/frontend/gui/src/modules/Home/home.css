.home-container {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  margin-left: 25%;
  margin-right: 25%;
}

.home-title {
  font-size: 50px;
  color: white;
}

.home-item {
  padding: 10px;
}

.home-button-container {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  align-content: space-between
}

.home-button-item {
  margin: 20px !important;
  height: 50px !important;
  width: 175px !important;
}