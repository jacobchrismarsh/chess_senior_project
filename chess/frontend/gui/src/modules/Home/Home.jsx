import React, { Component } from "react";
import { Square } from "../Game/subcomponents/Square";
import whitePawn from "../../cburnett/wp.svg";
import Button from '@material-ui/core/Button';
import autoBind from 'react-autobind';

import "../Dashboard/dashboard.css";
import "./home.css";

class Home extends Component {
  constructor(props) {
    super(props);

    autoBind(this);
  }

  render() {
    return (
      <div className="home-container">
        <br/>
        <br/>
        <div className="home-item home-title">Online Chess</div>
        <div className="home-item"><Square piece={whitePawn} /></div>
        <br/>
        <div className="home-item home-button-container">
          <Button
            size="medium"
            variant="contained"
            color="primary"
            className="chess-button home-button-item"
            onClick={() => window.location = '/sign_in'}
            >
            Sign In
          </Button>
          <div></div>
          <div></div>
          <Button
            size="medium"
            variant="contained"
            color="primary"
            className="chess-button home-button-item"
            onClick={() => window.location = '/create_user'}
          >
            Create Account
          </Button>
        </div>
      </div>
    );
  }
}

export default Home;