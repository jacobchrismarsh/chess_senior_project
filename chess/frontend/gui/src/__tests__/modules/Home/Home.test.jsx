import React from "react";
import { shallow } from 'enzyme';
import Home from "../../../modules/Home/Home";

describe("modules/Home/Home.jsx", () => {
  it('should render the Home component', () => {
    let home = shallow(<Home />);
    expect(home).toMatchSnapshot();
  });
});
