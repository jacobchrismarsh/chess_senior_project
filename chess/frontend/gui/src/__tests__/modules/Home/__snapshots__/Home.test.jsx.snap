// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules/Home/Home.jsx should render the Home component 1`] = `
<div
  className="home-container"
>
  <br />
  <br />
  <div
    className="home-item home-title"
  >
    Online Chess
  </div>
  <div
    className="home-item"
  >
    <Square
      piece="wp.svg"
    />
  </div>
  <br />
  <div
    className="home-item home-button-container"
  >
    <WithStyles(ForwardRef(Button))
      className="chess-button home-button-item"
      color="primary"
      onClick={[Function]}
      size="medium"
      variant="contained"
    >
      Sign In
    </WithStyles(ForwardRef(Button))>
    <div />
    <div />
    <WithStyles(ForwardRef(Button))
      className="chess-button home-button-item"
      color="primary"
      onClick={[Function]}
      size="medium"
      variant="contained"
    >
      Create Account
    </WithStyles(ForwardRef(Button))>
  </div>
</div>
`;
